ts/components/Collapsible.svelte

```diff
@@ -0,0 +1,73 @@
+<!--
+Copyright: Ankitects Pty Ltd and contributors
+License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
+-->
+<script lang="ts">
+    import { promiseWithResolver } from "../lib/promise";
+
+    export let id: string | undefined = undefined;
+    let className: string = "";
+    export { className as class };
+
+    export let collapsed = false;
+    let height = 0;
+
+    const [element, elementResolve] = promiseWithResolver<HTMLElement>();
+    let isCollapsed = false;
+
+    let style: string;
+    const setStyle = (el: HTMLElement) => {
+        height = el.clientHeight;
+        style = `--collapse-height: -${height}px; --duration: ${Math.sqrt(
+            height * 80,
+        )}ms`;
+    };
+
+    const transition = async (collapse: boolean) => {
+        const inner = await element;
+        isCollapsed = true;
+
+        if (collapse) {
+            setStyle(inner);
+
+            inner.addEventListener(
+                "transitionend",
+                () => {
+                    // DOM-manipulation is required here
+                    inner.setAttribute("hidden", "");
```

Comment:
The issue I faced with Svelte transitions is that they're only triggered when an element enters or leaves the DOM (i.e. with `{#if}` blocks). `Collapsible` wouldn't be a safe-to-use component if it had to destroy and re-mount its contents.

I tried it before (https://github.com/ankitects/anki/pull/2002/commits/9a2b3410d0ead37ae1da408d68e14507a058a613) and it lead to this error @dae mentioned in a [comment above](https://github.com/ankitects/anki/pull/2002#issuecomment-1212577220):

> JS error /_anki/js/reviewer_editor.js:78218 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'refresh')

because an array used in `NoteEditor.svelte` got shorter with each toggle until there were no elements left.